src/component/page/signin/index.tsx

```diff
@@ -1,37 +1,31 @@
-import { useAuth0 } from "@auth0/auth0-react";
+import { RedirectLoginOptions, useAuth0 } from "@auth0/auth0-react";
+import { Location } from "history";
+import { useLocation } from "react-router-dom";
 
 import {
   DefaultSigninProps,
   PlasmicSignin,
 } from "component/plasmic/shared/PlasmicSignin";
-import { useQuery } from "module/helpers";
 
 interface SigninProps extends DefaultSigninProps {}
 
 function Signin(props: SigninProps) {
+  const { state } = useLocation() as Location<{ returnTo?: string } | null>;
   const { loginWithRedirect } = useAuth0();
-  const query = useQuery();
 
-  const ventureId = query.get("ventureId") ?? "";
-  const timelineId = query.get("timelineId") ?? "";
-  const code = query.get("code") ?? "";
-  const id = query.get("id") ?? "";
-  const ventureName = query.get("ventureName") ?? "";
-  const resource = query.get("resource") ?? "";
-  const role = query.get("role") ?? "";
-
-  localStorage.setItem("ventureId", ventureId);
-  localStorage.setItem("timelineId", timelineId);
-  localStorage.setItem("code", code);
-  localStorage.setItem("id", id);
-  localStorage.setItem("ventureName", ventureName);
-  localStorage.setItem("resource", resource);
-  localStorage.setItem("role", role);
+  let redirectLoginOptions: RedirectLoginOptions | undefined = undefined;
+  if (state?.returnTo) {
+    redirectLoginOptions = {
+      appState: {
+        returnTo: state.returnTo,
```

Comment:
Save originally requested page from router state to auth0 state. Then it will be accessible after authentication succeeds.